### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by Ngufor-emmanuella**
### In this project @Samuellekpor and @Ngufor-emmanuella worked on the following 
Login page and registration page:
Should be built with Devise.
Food list:
Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
Should lead to a form that allows users to add new food.
Recipes list:
Should display a list of recipes created by the logged-in user as in the wireframe.
Should lead to recipe details.
If the user is the owner of the recipe, should allow the user to delete it.
Public recipe list:
Should display a list of all public recipes ordered by newest as in the wireframe.
Should lead to recipe details.
Recipe details:
If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
General shopping list view (only for a group that has less than 3 members):
Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
Should count the total food items and total price of the missing food.
Make sure there are no N+1 queries happening.
Create a navigation menu that allows users to open all of the pages you created.
Technical Setup

Set up Devise for authentication.
Set up RSpec and Capybara testing libraries.

### CODE REVIEW

File: app/views/foods/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        <th scope="col">Actions</th>
+      </tr>
+    </thead>
+    <% @foods.each do |food| %>
+    <tbody>
+      <tr>
+        <th scope="row"><%= food.id %></th>
+        <td><%= food.name %></td>
+        <td><%= food.measurement_unit %></td>
+        <td><%= food.price %></td>
```
Comment (zieeco) [inline app/views/foods/index.html.erb]: - Hi Team, please, let's kindly add the missing $ sign to our price as this indicates it price in dollar to match the design as much as we can. See the image below:

*wireframe*
Screenshot from 2023-03-04 14-50-53
*your design*
Screenshot from 2023-03-04 14-50-10

File: app/views/recipe_foods/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<div class='new_container'>
+  <h1>Add Ingredients</h1>
+  <%= form_with model: @recipe_food, url: user_recipe_recipe_foods_path(params[:user_id], params[:recipe_id]), local: true, method: :post, scope: :recipe_food do |f| %>
+    <%= f.number_field :quantity, placeholder: "Quantity", rows: "8", required: true %>
+    <div class="form-group">
+    <%= f.select :food_id, @foods.map { |t| [t.name, t.id]}, {}, class: "form-control" %>
+    </div>
+    <%= f.submit "Add food", class: 'button' %>
```
Comment (zieeco) [inline app/views/recipe_foods/new.html.erb]: - Hi Team, when trying to add an ingredient, here's the error encountered. please, let's fix this


Screenshot from 2023-03-04 15-10-43


- After fixing this error, we should be able to generate the shopping list as expected and it should match the wireframe.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,223 @@
...
+
+> Ruby on Rails basic project to build a structure for recipes, recipes details and where the user can login and then create a recipe accordingly.
+
+**[Recipes-app]** Ruby on Rails basic project to build a structure for recipes, recipes details and where the user can login and then create a recipe accordingly.
+
+## 🛠 Built With <a name="built-with"></a>
+
+### Tech Stack <a name="tech-stack"></a>
+
+> Describe the tech stack and include only the relevant sections that apply to your project.
```
Comment (zieeco) [inline README.md]: - Hi, please, let's kindly remove these placeholder texts as these are guides to help you update the contents of your readme. it's expected to be removed after you've updated the readme to make it more professional.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,223 @@
...
+- **[Recipes lists]**
+- **[Shopping lists]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+> Add a link to your deployed project.
```
Comment (zieeco) [inline README.md]: - Hi, please, let's kindly remove these placeholder texts as these are guides to help you update the contents of your readme. it's expected to be removed after you've updated the readme to make it more professional.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,223 @@
...
+
+- [Live Demo Link]()
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- GETTING STARTED -->
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+> Describe how a new developer could make use of your project.
```
Comment (zieeco) [inline README.md]: - Hi, please, let's kindly remove these placeholder texts as these are guides to help you update the contents of your readme. it's expected to be removed after you've updated the readme to make it more professional.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,223 @@
...
+  `rails server`
+
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- AUTHORS -->
+
+## 👥 Authors <a name="authors"></a>
+
+> Mention all of the collaborators of this project.
```
Comment (zieeco) [inline README.md]: - Hi, please, let's kindly remove these placeholder texts as these are guides to help you update the contents of your readme. it's expected to be removed after you've updated the readme to make it more professional.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,223 @@
...
+- GitHub: [Samuel lekpor](https://github.com/Samuel-lekpor)
+- LinkedIn: [Samuel lekpor](https://www.linkedin.com/in/Samuel-lekpor/)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- FUTURE FEATURES -->
+
+## 🔭 Future Features <a name="future-features"></a>
+
+> Describe 1 - 3 features you will add to the project.
```
Comment (zieeco) [inline README.md]: - Hi, please, let's kindly remove these placeholder texts as these are guides to help you update the contents of your readme. it's expected to be removed after you've updated the readme to make it more professional.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,223 @@
...
+
+Feel free to check the [issues page](../../issues/).
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- SUPPORT -->
+
+## ⭐️ Show your support <a name="support"></a>
+
+> Write a message to encourage readers to support your project
```
Comment (zieeco) [inline README.md]: - Hi, please, let's kindly remove these placeholder texts as these are guides to help you update the contents of your readme. it's expected to be removed after you've updated the readme to make it more professional.

File: README.md [hunk 9]
```diff
@@ -1,2 +1,223 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
+
+_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
```
Comment (zieeco) [inline README.md]: - Hi, please, let's kindly remove these placeholder texts as these are guides to help you update the contents of your readme. it's expected to be removed after you've updated the readme to make it more professional.

File: README.md [hunk 10]
```diff
@@ -1,2 +1,223 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+> Add a link to your deployed project.
+
+- [Live Demo Link]()
```
Comment (zieeco) [inline README.md]: - The live demo link is broken, since you've not deployed your app, kindly remove it to make the readme more professional.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Team,


Great job so far! 👏 
You've done well with the recipe app, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️
- Good commit message :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

- [ ] Hey, some of your tests are failing about a 14 test chases fails. please, can you check this to fix it? See the image below:
Screenshot from 2023-03-04 15-44-14
- [ ] Models test passed
- [ ] Request fails
- [ ] Integration fails
- [ ] views fails






## Optional suggestions
_Check the comments under the review._

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: .gitattributes [hunk 11]
```diff
@@ -0,0 +1,7 @@
+# See https://git-scm.com/docs/gitattributes for more about git attribute files.
+
```
Comment (MeqdamAlqudah) [inline .gitattributes:2]: - [ ] Please I'm still getting the same error as the last reviewer's notice when the user tries to add ingredients.
image

Comment (MeqdamAlqudah) [inline .gitattributes:2]: - [ ] Good job by resolving the failed tests as the last reviewer said but there is still one failed test please fix it.
image

**Review by MeqdamAlqudah — CHANGES_REQUESTED** [review]
Hi @Ngufor-emmanuella , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
image


 ## Highlights
:+1: no linter errors 
:+1: nice design

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Deepakdanger — APPROVED** [review]
Hi @Ngufor-emmanuella & @Samuellekpor, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**